src/kit/internal/Elevation.module.scss

```diff
@@ -0,0 +1,57 @@
+/* stylelint-disable order/order */
+/* stylelint-disable no-descending-specificity */
+@use 'sass:map';
+@use 'sass:meta';
+@use 'sass:math';
+
+$min-elevation: 0;
+$max-elevation: 4;
+$levels: (
+  0: var(--theme-elevation-0-bg),
+  1: var(--theme-elevation-1-bg),
+  2: var(--theme-elevation-2-bg),
+  3: var(--theme-elevation-3-bg),
+  4: var(--theme-elevation-4-bg),
+);
+$level-highlights: (
+  0: var(--theme-elevation-0-hover),
+  1: var(--theme-elevation-1-hover),
+  2: var(--theme-elevation-2-hover),
+  3: var(--theme-elevation-3-hover),
+  4: var(--theme-elevation-4-hover),
+);
+$border-colors: (
+  0: var(--theme-elevation-0-border),
+  1: var(--theme-elevation-1-border),
+  2: var(--theme-elevation-2-border),
+  3: var(--theme-elevation-3-border),
+  4: var(--theme-elevation-4-border),
+);
+
+@mixin generate-variables {
+  @for $level from $min-elevation through $max-elevation {
+    .elevation-#{$level} {
+      --theme-elevation-current-bg: #{map.get($levels, $level)};
+      --theme-elevation-current-border: #{map.get($border-colors, $level)};
+      --theme-elevation-current-bg-highlight: #{map.get($level-highlights, $level)};
+    }
+  }
+}
+
+@mixin elevation {
+  background-color: var(--theme-elevation-current-bg);
+  border-color: var(--theme-elevation-current-border);
+
+  &.hover:hover {
+    background-color: var(--theme-elevation-current-bg-highlight);
+  }
+  &.border {
+    border: var(--theme-stroke-width) solid var(--theme-elevation-current-border);
+  }
+}
```

Comment:
do the hover and border selectors added by this mixin work for the pivot component? I know we don't want it there, but i'm wondering if the `.hover` and `.border` classes we're creating here are hashed here or in the place they're included, because the elevation hook assumes the hash will be the same.

Reply:
Hm, it seems like they don't. I'm looking into how to make that work, but if you have ideas feel free to let me know.

Reply:
I took the border and hover options out of `useElevation` for now because I couldn't figure out a good way to handle it.

Reply:
that seems like the best option -- you could have the mixin take the `hover` and `border` options instead, but it wouldn't work for the (currently only) place where we'd use it because there's no place to pass the options from the component to the css. It might be a little confusing for the mixin to add classes that aren't accessible, though. I would probably nest the hover and border selectors under the elevationBase selector.

Reply:
Could you use `:local()` to just make the hover and border classes not hashed?